alacritty/src/event.rs

```diff
@@ -495,6 +497,38 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
         }
     }
 
+    /// Update the cursor blinking state.
+    #[inline]
+    fn update_cursor_blinking(&mut self) {
+        // Get config cursor style.
+        let mut cursor_style = self.config.cursor.style;
+        if self.terminal.mode().contains(TermMode::VI) {
+            cursor_style = self.config.cursor.vi_mode_style.unwrap_or(cursor_style);
+        };
+
+        // Check terminal cursor style.
+        let blinking = cursor_style.blinking(self.terminal.cursor_style().blinking);
```

Comment:
I'm not sure if I like this API. The `cursor_style.blinking(current_state)` thing is a bit weird, but that's the easiest API for us in Alacritty.

I think the other alternative would be a bit nicer, which would be `cursor_style.blinking_override().unwrap_or(terminal_state)`, but it's not as concise for our usage.

What do you think @kchibisov?